core/Pos/Core/Update/Types.hs

```diff
@@ -296,6 +299,23 @@ mkSystemTag tag | T.length tag > systemTagMaxLength
                 | otherwise
                     = pure $ SystemTag tag
 
+-- | @SystemTag@ corresponding to the operating system/architecture pair the program was
+-- compiled in.
+-- The @System.Info@ module from @base@ was used
+-- (https://hackage.haskell.org/package/base-4.10.1.0/docs/src/System.Info.html#os)
+-- but the @os@ and @arch@ values are @String@'s. For an exhaustive list of possible
+-- platforms, see
+-- https://downloads.haskell.org/~ghc/8.0.1/docs/html/libraries/Cabal-1.24.0.0/Distribution-System.html.
+currentSystemTag :: SystemTag
+currentSystemTag =
+    let f sys | sys == "mingw32" = "win"
+              | sys == "osx"     = "mac"
+              | otherwise      = sys
+        g archt | archt == "x86_64" = "64"
+                | archt == "i386"   = "32"
+                | otherwise         = archt
+    in $(runQ [| SystemTag (toText (f os ++ g arch)) |])
```

Comment:
Currently we expect it to be `macos64` or `win64` or something for linux. I think it'd be good to add a test that `currentSystemTag` is one of these values. In this case we'll notice if it changes for some reason.